cmake/api_macros.cmake

```diff
@@ -18,11 +18,13 @@ macro(OPENDDS_GET_SOURCES_AND_OPTIONS
   idl_prefix
   libs
   tao_options
-  opendds_options)
+  opendds_options
+  link_keyword_option)
 
   set(_options_n
     PUBLIC PRIVATE INTERFACE
-    TAO_IDL_OPTIONS OPENDDS_IDL_OPTIONS)
+    TAO_IDL_OPTIONS OPENDDS_IDL_OPTIONS
+    TARGET_LINK_KEYWORD)
```

Comment:
I am thinking it might actually be better not to allow users to specify the scope because linking it against the Threads::Threads is an implementation detail. For instance, it does not make sense to have Threads::Threads as a transitive dependency (PUBLIC scope) as only a target which is passed to OPENDDS_TARGET_SOURCES needs such a dependency and hence, PRIVATE scope makes more sense.

Reply:
Yes, that would be the simplest solution if it was possible, but references to our dependencies can escape through inlined code.

Reply:
Not sure I understand that scenario. Any way you might provide an example? If that’s the case, then is there any reason for it to be PRIVATE? In other words, we could just use PUBLIC to cover all scenarios and avoid complicating the OPENDDS_TARGET_SOURCES interface.

Reply:
My initial approach was to simply use PUBLIC. When the cmake_integration tests failed, however, because the test source made use of the plain signature, I realised that most users would likely not be familiar with  using linking scope, and that their projects would produce cmake errors until they comply with the linking signature policy.

That being said, it would probably be less effort to for a user to read up on the error and apply the relevant linking scope to their target, than to try understand motivation behind the new  OPENDDS_TARGET_SOURCES signature and  perhaps make a guess as to which keyword to specify for the linking.

Reply:
Ah, okay. Given that the old [CMP0023 ](https://cmake.org/cmake/help/latest/policy/CMP0023.html) behavior is deprecated, it seems the tests should be updated and within the [OpenDDSConfig.cmake](https://github.com/objectcomputing/OpenDDS/blob/master/cmake/OpenDDSConfig.cmake#L68) we need to have a line forcing users to use non-deprecated behavior:

`cmake_policy(SET CMP0023 NEW)`